### PR TITLE
[add][lg] create methods and test - #4

### PR DESCRIPTION
**Pull request by mariorodeghiero**
- create action function to be able create/update dataset and resource
- add test to create a dataset
- add test to create/update a resource
- add test to the action function

**Note:** Please check more details on how it works in `test/metadata.test.js`

I also did a **real** test like the code below and it works with ckan:

```
const client = new Client('XXXX-XXXX-XXXX-XXXX', 'myorg', 'data-test-2', 'http://localhost:5000')

client.put('package_create', { "name": "my_dataset_4", "owner_org": "myorg"})
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -29,6 +30,11 @@ class Client {
     throw "invalid URL argument"
   }
 
+  async put(actionType, data) {
```
Comment (anuveyatsu) [inline lib/index.js:33]: Do we need `async` here if we aren't using `await` in this function/method?

File: lib/index.js [hunk 2]
```diff
@@ -29,6 +30,11 @@ class Client {
...
+    const response = ActionApi.action(this.api, this.authToken, actionType, data)
+    return response
```
Comment (anuveyatsu) [inline lib/index.js:35]: Or  we can `return await response`

Comment (anuveyatsu) [reply lib/index.js:35]: I'd even shorten this code to `return ActionApi.action(this.api, this.authToken, actionType, data)`

File: package.json [hunk 3]
```diff
@@ -27,7 +27,8 @@
   "license": "MIT",
   "dependencies": {
     "axios": "^0.19.2",
-    "crypto-js": "^4.0.0"
+    "crypto-js": "^4.0.0",
+    "data.js": "^0.12.11"
```
Comment (anuveyatsu) [inline package.json:31]: Looks like we're using this only for the tests so maybe it can be a dev dependency?

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
Great stuff! Please see my comments for minor improvements.

File: lib/index.js (hunk 2 above)
Comment (mariorodeghiero) [reply lib/index.js:35]: yes, sounds better.

File: lib/index.js (hunk 1 above)
Comment (mariorodeghiero) [reply lib/index.js:33]: Probably not, let do a double-check with calm.

File: package.json (hunk 3 above)
Comment (mariorodeghiero) [reply package.json:31]: I need to check with @rufuspollock  if he wants to use "data.js" to do something into the SDK. Is not clear for me atm

File: lib/index.js (hunk 2 above)
Comment (mariorodeghiero) [reply lib/index.js]: Done! now is:

`return ActionApi.action(this.api, this.authToken, actionType, data)`

Real test keep working, see the response below:

```
node index.js
test:  {
  help: 'http://localhost:5000/api/3/action/help_show?name=package_create',
  success: true,
  result: {
    license_title: null,
    maintainer: null,
    relationships_as_object: [],
    private: false,
    maintainer_email: null,
    num_tags: 0,
    id: 'd0cc5340-bfd4-47e6-a41c-94233d14a203',
    metadata_created: '2020-09-11T12:24:35.576884',
    metadata_modified: '2020-09-11T12:24:35.576889',
    author: null,
    author_email: null,
    state: 'active',
    version: null,
    creator_user_id: 'ceb0d8c2-352b-4a6c-b2c8-2a1a4190192b',
    type: 'dataset',
    resources: [],
    num_resources: 0,
    tags: [],
    groups: [],
    license_id: null,
    relationships_as_subject: [],
    organization: {
      description: 'my org',
      created: '2020-04-14T13:27:47.434967',
      title: 'myorg',
      name: 'myorg',
      is_organization: true,
      state: 'active',
      image_url: '',
      revision_id: 'bff67202-a21e-470b-9b67-14e65ad85fa2',
      type: 'organization',
      id: 'f3fce98a-4750-4d09-a5d1-d5e1b995a2e8',
      approval_status: 'approved'
    },
    name: 'my_dataset_5',
    isopen: false,
    url: null,
    notes: null,
    owner_org: 'f3fce98a-4750-4d09-a5d1-d5e1b995a2e8',
    extras: [],
    title: 'my_dataset_5',
    revision_id: '19b38438-4d26-46f8-ae8b-bef7e7e7cafc'
  }
}
```

File: lib/index.js (hunk 1 above)
Comment (mariorodeghiero) [reply lib/index.js]: @anuveyatsu  Done, Removed!
The real test result in the comment below.

**Review by anuveyatsu — APPROVED** [review]
LGTM

**Pull request merged**